backend/src/LibExecution/Interpreter.fs

```diff
@@ -628,23 +630,42 @@ and callFn
               DError(sourceID, $"Function {FQFnName.toString desc} is not found")
 
         | Some fn ->
-          // equalize length
-          let expectedLength = List.length fn.parameters in
-          let actualLength = List.length argvals in
+          // ensure we have the expected # of typeArguments _and_ arguments values
+          let expectedTypeParamLength = List.length fn.typeArgs
+          let expectedArgLength = List.length fn.parameters
+
+          let actualTypeArgLength = List.length typeArgs
+          let actualArgLength = List.length argvals
+
+          let err errMsg = DError(sourceID, errMsg)
 
-          if expectedLength = actualLength then
+          match (expectedTypeParamLength = actualTypeArgLength),
```

Comment:
TODO: add test cases around all 3 error messages (currently, only the arg # mismatch is tested).